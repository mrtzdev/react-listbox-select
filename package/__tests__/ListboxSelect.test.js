import React from "react";
import renderer from "react-test-renderer";

import { ListboxSelect } from "../index";

describe("ListboxSelect renders correctly", () => {
  test("snapshot renders", () => {
    const options = [
      { label: "Grapes", value: "grapes" },
      { label: "Mango", value: "mango" },
      { label: "Strawberry", value: "strawberry" },
    ];

    const component = renderer.create(
      <ListboxSelect
        title="Single Listbox Select"
        list={options}
        value={{ label: "Mango", value: "mango" }}
        className="my-class-name"
        prefixId="SingleSelect"
        collapsible={true}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
