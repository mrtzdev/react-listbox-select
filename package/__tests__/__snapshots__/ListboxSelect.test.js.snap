// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListboxSelect snapshot renders 1`] = `
Array [
  <div
    className="react-listbox-select my-class-name"
  >
    <div
      className="listbox-title clickable  "
      id="rlsm_elem_SingleSelect"
      onClick={[Function]}
    >
      Single Listbox Select
    </div>
    <div
      className="listbox-container collapsible "
    >
      <div>
        <ul
          aria-labelledby="rlsm_elem_SingleSelect"
          className="list "
          id="rlsm_elem_list_SingleSelect"
          role="listbox"
        >
          <li
            aria-label="Grapes"
            aria-selected="false"
            className="item "
            id="rlsm_elem_SingleSelect_grapes"
            role="option"
          >
            <button
              aria-label="Grapes"
              className="btn-filter"
              onClick={[Function]}
            >
              <div
                className="filter-item"
              >
                <span
                  className="filter-label"
                >
                   
                  Grapes
                </span>
              </div>
            </button>
          </li>
          <li
            aria-label="Mango"
            aria-selected="false"
            className="item "
            id="rlsm_elem_SingleSelect_mango"
            role="option"
          >
            <button
              aria-label="Mango"
              className="btn-filter"
              onClick={[Function]}
            >
              <div
                className="filter-item"
              >
                <span
                  className="filter-label"
                >
                   
                  Mango
                </span>
              </div>
            </button>
          </li>
          <li
            aria-label="Strawberry"
            aria-selected="false"
            className="item "
            id="rlsm_elem_SingleSelect_strawberry"
            role="option"
          >
            <button
              aria-label="Strawberry"
              className="btn-filter"
              onClick={[Function]}
            >
              <div
                className="filter-item"
              >
                <span
                  className="filter-label"
                >
                   
                  Strawberry
                </span>
              </div>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  " ",
]
`;
